src/Chainweb/Pact/PactService.hs

```diff
@@ -622,21 +624,64 @@ execNewGenesisBlock miner newTrans = withDiscardedBatch $
 execLocal
     :: CanReadablePayloadCas tbl
     => ChainwebTransaction
-    -> PactServiceM tbl (P.CommandResult P.Hash)
-execLocal cmd = withDiscardedBatch $ do
+    -> Maybe LocalPreflightSimulation
+      -- ^ preflight flag
+    -> Maybe LocalSignatureVerification
+      -- ^ turn off signature verification checks?
+    -> Maybe BlockHeight
```

Comment:
It's slightly confusing to have `BlockHeight` (which has absolute vibes) used as a delta (which has relative vibes). Maybe just another couple words of documentation could help.

Or more drastically, changing the parameter to be a function `(BlockHeight -> BlockHeight)`, and the user would pass `id` to indicate they want no rewinding, and `const (BlockHeight 10)` to indicate they want to play from block height 10.

i guess the case I'm worrying about is someone seeing the `BlockHeight` type and passing `Just (BlockHeight 1000)` to try to play from block height 1000. Confusing a relative value for an absolute one.

Reply:
That's a good point. We're not rewinding to a block height, but to a block height delta. When we configure the checkpointer to execute for a rewind, it cares about a blockheight delta defined to be blockheights to be subtracted from the current height. We don't really have a good language for expressing this. I would probably reject the idea that it should be a function, since it's always going to be an override if it's a positive height (it's either `id` or `\_ -> rewindDepth`, with no in between). So perhaps this warrants a newtype a la 

```haskell
newtype RewindDepth = RewindDepth BlockHeight
  deriving ... via BlockHeight
```

It can just exist with the other local data types. However, at this point i think we're kind of getting into defining a universe of types specific to the theory of mind for what a particular endpoint intends to do with existing datatypes rather than defining concretely different things. If we're okay with doing that, I'd want to follow up this PR with a PR that changes it for the Checkpointer as well, since the `BlockHeight` representation here percolates up from the API defined for the Checkpointer.

Reply:
Addressed in DM